ReSharper.FSharp/src/FSharp.Psi.Features/src/CodeCleanup/FantomasSettingsConverterProvider.fs

```diff
@@ -0,0 +1,43 @@
+module JetBrains.ReSharper.Plugins.FSharp.Psi.Features.CodeCleanup.FSharpEditorConfig
+
+open System
+open JetBrains.Application
+open JetBrains.Application.Settings
+open JetBrains.Application.Settings.Calculated.Implementation
+open JetBrains.DataFlow
+open JetBrains.Diagnostics
+open JetBrains.Lifetimes
+open JetBrains.ReSharper.Plugins.FSharp.Services.Formatter
+open JetBrains.ReSharper.Psi.EditorConfig
+
+type FantomasSettingsConverter(fantomasSettingsEntry: SettingsIndexedEntry) =
+  let fSharpPrefix = "fsharp_"
+
+  let run (context: SettingsConvertContext<'a>) _ =
+      let context = context.As<SettingsConvertContext<string>>()
+      Assertion.AssertNotNull(context, "Context expected to be SettingsConvertContext<string>")
+
+      for entry in context.SourceData.FindEntriesByPrefix(fSharpPrefix) do
+        let setting = SettingIndex(fantomasSettingsEntry, entry.Key)
+        context.Target.SetValue(setting, entry.Value)
+
+      true
+
+  interface IEditorConfigConverter with
+    member val Category = "F#" with get, set
+    member x.IsPropertySupported(property) = property.StartsWith(fSharpPrefix, StringComparison.Ordinal)
+    member x.Convert(context) = run context null
+    member x.ConvertAndCheck(context) = run context context
+    member x.ReverseConvert _ = failwith "Not implemented"
+
+
+[<ShellComponent>]
+type FantomasSettingsConverterProvider(lifetime: Lifetime, schema: SettingsSchema) =
+    let items = CollectionEvents<IEditorConfigConverter>(lifetime, $"{nameof FantomasSettingsConverterProvider}.Items")
+
+    do
+      let fantomasSettingsEntry = schema.GetIndexedEntry<_, _>(fun (key: FSharpFormatSettingsKey) -> key.FantomasSettings)
+      items.Add(FantomasSettingsConverter(fantomasSettingsEntry))
```

Comment:
Is this conversion really needed at the shell startup? Components in this container are expected not to do heavy work unless really required, since every component initialization slows down the startup.

Reply:
There is no conversion here, it's just a creating of EditorConfig schema. Yep, it happens in Shell as all settings schemas, but don't think it will slow something down.

Reply:
@osdm I see, thanks!